lib/gradebook.rb

```diff
@@ -1 +1,52 @@
+class Gradebook
+  attr_reader :instructor, :courses, :all_students
+  def initialize (instructor)
+    @instructor = instructor
+    @courses = []
+    @all_students_a
+  end
 
+  def add_course(course)
+    @courses << course
+  end
+
+  def all_students
+    all_students = []
+    @courses.each {|course| all_students << course.students}
+    all_students.flatten
+  end
+
+  def list_all_students
+    all_students_h = Hash.new {|hash, key| hash[key] = []}
+    @courses.each do |course|
+      all_students_h[course.name] = course.students
+    end
+    all_students_h
+  end
+
+  def students_below(score)
+    score = score.to_f
+    all_students = self.all_students
+    failing_students = all_students.select do |student|
+      student.grade < 70
```

Comment:
According to the interaction pattern, this should use the dynamic threshold passed in as an argument:
```suggestion
      student.grade < score
```